internal/app/upstream/client_test.go

```diff
@@ -27,6 +30,16 @@ func TestOpenStreamShouldReturnErrorForInvalidTypeUrl(t *testing.T) {
 	assert.Nil(t, respCh)
 }
 
+func TestOpenStreamShouldReturnErrorForInvalidTypeUrlV3(t *testing.T) {
```

Comment:
i had to copy these tests because the setup is not going well with BDD style testing. I could have split the tests into v2 and v3 files, but preferred this way for readability. Let me know if you feel we should split the tests into v2 and v3 files.

Reply:
I think how you structured this is OK since the client impl is shared between v2 and v3.